src/python/pants/goal/run_tracker.py

```diff
@@ -503,3 +518,18 @@ def _get_option_to_record(self, scope, option=None):
             raise ValueError(
                 f"Couldn't find option scope {scope}{option_str} for recording ({e!r})"
             )
+
+    def retrieve_logs(self) -> List[str]:
+        """Get a list of every log entry recorded during this run."""
+
+        if not self.run_logs_file:
+            return []
+
+        output = []
+        try:
+            with open(self.run_logs_file, "r") as f:
+                output = f.readlines()
```

Comment:
Nit: you can use `Path(self.run_logs_file).read_text()` for a one-liner.

Reply:
from a plugin author POV, I would much rather get the filename...
I might just end up accessing `run_tracker. run_logs_file` directly and won't use this API at all...
since my plan is to stream the log, so loading it into memory and copying it multiple times is a bad idea.